### ISSUE TITLE
commenting out the second if statement returns a failed test. The test should pass as the commented out if statement has no bearing on the return statement. 

### ISSUE THREAD

**Issue by Aldon2** [FreeCodeCamp/FreeCodeCamp#10745]
Challenge [Comparisons with the Logical And Operator](https://www.freecodecamp.com/challenges/comparisons-with-the-logical-and-operator#?solution=%0Afunction%20testLogicalAnd(val)%20%7B%0A%20%20%2F%2F%20Only%20change%20code%20below%20this%20line%0A%0A%20%20if%20(val%3C%3D50%20%26%26%20val%3E%3D25)%20%7B%0A%20%20%20%20%2F%2Fif%20(val)%20%7B%0A%20%20%20%20%20%20return%20%22Yes%22%3B%0A%20%20%20%20%7D%0A%20%20%2F%2F%20Only%20change%20code%20above%20this%20line%0A%20%20return%20%22No%22%3B%0A%7D%0A%0A%2F%2F%20Change%20this%20value%20to%20test%0AtestLogicalAnd(10)%3B%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.101 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:
```javascript

function testLogicalAnd(val) {
  // Only change code below this line

  if (val<=50 && val>=25) {
    //if (val) {
      return "Yes";
    }
  // Only change code above this line
  return "No";
}

// Change this value to test
testLogicalAnd(10);

```

**Comment by Aldon2** [comment]
So as with the code above the if statement is commented out but returns a failed test saying that there should only be one if statement. once you remove the commented out if statement without any other code changes all tests pass. So the test is evaluating a comment as if it was a statement which it is not!

**Comment by ryzokuken** [comment]
Evidently, the code counting the number of `if`s does not pay any regard to the fact if it is commented or not, it's just plain old counting the number of `if`s present inside your code.

**Comment by Aldon2** [comment]
Yes exactly! thank you for looking into it! How would we go about doing that? Can I contribute to fixing it?

**Comment by ryzokuken** [comment]
@Aldon2: You certainly could. The current code is pretty simplistic, it finds the number of `if`s in the code. To make it skim over commented portion, you'd need some pretty serious regexp, analysing comments using `//` - `\n` and `/*` - `*/` pairs. Try putting your algorithmic thinking to use.

**Comment by thewhisperinyourears** [comment]
But what is the else here and the val being 10 is never caught in the if

Admin staff

**Comment by ryzokuken** [comment]
@thewhisperinyourears I don't understand what you mean... if the condition is false, it returns `"No"`

**Comment by Bouncey** [comment]
A PR to fix this will be raised against #11360
